security/sec-iride/src/main/java/org/geoserver/security/iride/IrideRoleService.java

```diff
@@ -95,6 +95,8 @@
     private HttpClient httpClient = new HttpClient();
     private HttpConnectionManagerParams params = new HttpConnectionManagerParams();
 
+    private String username;

```

Comment:
You can't do this, this needs to work in multithread environment, each thread (each request) can use tthe same instance of the service, but the username is different, so it cannot be stored as a service field